zen/scene/view.c

```diff
@@ -26,6 +26,36 @@ zn_view_is_mapped(struct zn_view *self)
   return !wl_list_empty(&self->link);
 }
 
+void
+zn_view_focus(struct zn_view *self)
+{
+  struct zn_server *server = zn_server_get_singleton();
+  struct zn_scene *scene = server->scene;
+
+  if (scene->active_view != NULL) {
+    zn_view_unfocus(scene->active_view);
+  }
+
+  if (self != NULL) {
+    self->impl->focus(self);
+  }
```

Comment:
We should not assume `self` is NULL
```suggestion
  self->impl->focus(self);
```